cli/src/commands/init/index.ts

```diff
@@ -1,17 +1,292 @@
+import path from 'path';
+import prompts from 'prompts';
+import which from 'which';
 import { BaseCommand } from '../../base.js';
+import { readAPIKey } from '../../key.js';
+import { xataDatabaseSchema } from '../../schema.js';
+import Codegen from '../codegen/index.js';
+import EditSchema from '../schema/edit.js';
 
 export default class Init extends BaseCommand {
-  static description = 'Initialize a database';
+  static description = 'Configure your working directory to work with a Xata database';
 
   static examples = [];
 
-  static flags = {};
+  static flags = {
+    databaseURL: this.databaseURLFlag,
+    schema: Flags.string({
+      description: 'Initializes a new database or updates an existing one with the given schema'
+    }),
+    force: Flags.boolean({
+      description: 'Overwrite existing project configuration'
+    })
+  };
 
-  static args = [];
+  static args = [
+    // TODO: add an arg for initial schema
+  ];
 
   async run(): Promise<void> {
-    // const { args, flags } = await this.parse(Init);
+    const { flags } = await this.parse(Init);
 
-    this.error('To be done');
+    if (this.projectConfigLocation) {
+      if (!flags.force) {
+        this.error(
+          `Project already configured at ${this.projectConfigLocation}. Use ${chalk.bold('--force')} to overwrite it`
+        );
+      } else {
+        this.warn(`Will overwrite ${this.projectConfigLocation} because ${chalk.bold('--force')} is being used`);
+      }
+    }
+
+    const { workspace, database, databaseURL } = await this.getParsedDatabaseURL(flags.databaseURL, true);
+
+    this.projectConfig = { databaseURL };
+
+    await this.installSDK();
+
+    await this.writeConfig();
+
+    await this.writeEnvFile();
+
+    if (flags.schema) {
+      // TODO: remove default value once the getCurrentBranchName() return type is fixed.
+      // It actually always returns a string, but the typing is wrong.
+      const branch = (await getCurrentBranchName()) || '';
+      await this.deploySchema(workspace, database, branch, flags.schema);
+    } else {
+      await EditSchema.run([]);
+    }
+
+    await Codegen.runIfConfigured(this.projectConfig);
+
+    this.log('Done. You are all set!');
+  }
+
+  async installSDK() {
+    const { confirm } = await prompts({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to install the TypeScript/JavaScript SDK?',
+      initial: true
+    });
+    if (!confirm) return;
+
+    await this.installPackage('@xata.io/client');
+
+    await this.configureCodegen();
+  }
+
+  async configureCodegen() {
+    const { confirm } = await prompts({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to use the TypeScript/JavaScript code generator?',
+      initial: true
+    });
+    if (!confirm) return;
+
+    this.projectConfig = this.projectConfig || {};
+    this.projectConfig.codegen = {};
+
+    const { output } = await prompts({
+      type: 'text',
```

Comment:
[idea] Would be cool that this allowed you to navigate existing folders, select existing file or create new file interactively instead of text path.

Reply:
Ohhh, nice idea. That'd be super cool. I'll see if there's a plugin for that, or we could try creating ours.